MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -20,13 +24,17 @@ final class FeedViewModel {
 	
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onFeedLoadError: Observer<LoadFeedErrorViewModel>?
```

`onFeedLoadError: Observer<LoadFeedErrorViewModel>?` could be `onErrorStateChange: Observer<String?>?`, so the ViewModel decides when to show/hide errors (instead of the view controller). It's often easier to manage state when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange?(.none)` to hide the error before loading the feed.

An on failure, the ViewModel can call  `onErrorStateChange?(Localized.Feed.loadError)` to show the localized error.

---

I thought about doing it that way first, but then I thought, wouldn't I be passing the logic on when to show the error to the viewController? Same as we do with `onLoadingStateChange ` then the VC has to check wether the errorMsg is nil or not, while in the other approach I command the vc to show or hide

      viewModel.onErrorStateChange = { error in 
             guard let errorMsg = error else {
                       errorView.hideErrorMsg()
                       return
              }
              errorView.show(errorMsg)
      }
      
It does make more sense to have it centralized in a single closure though.
Also what do you think about encapsulating the errorMsg in a struct in case I want to pass another param in the future? like an errorIcon